app/models/manageiq/providers/vmware/infra_manager/refresh_worker/runner.rb

```diff
@@ -44,6 +44,7 @@ def ensure_inventory_collector
     return if collector&.running?
 
     _log.warn("Inventory collector thread not running, restarting...") unless collector.nil?
+    stop_inventory_collector
```

Comment:
I don't understand how it can get here.  the `if collector&.running?` will check [`vim_thread&.alive?`](https://github.com/ManageIQ/manageiq-providers-vmware/pull/700/files#diff-1354967fd69afcb0caf8af2d735d4dc26246f72e37a195a456cc0608fbd184d0R26) ahead of it, so it won't call stop, which you've changed to again check `vim_thread&.alive?`

Reply:
So this is really `stop_saver_thread` at this point, kind of making my point that I think we need to refactor this a bit https://github.com/ManageIQ/manageiq-providers-vmware/pull/700#issuecomment-776918286

If the vim_thread isn't alive it will get here.  It will stop the vim_thread if it is alive (which it isn't) then stop the saver thread (which is what we actually need to do)